src/components/button-toggle/button-toggle.ts

```diff
@@ -0,0 +1,341 @@
+
+  constructor(@Optional() toggleGroup: MdButtonToggleGroup,
+              @Optional() toggleGroupMultiple: MdButtonToggleGroupMultiple,
+              public buttonToggleDispatcher: MdUniqueSelectionDispatcher) {
+    this.buttonToggleGroup = toggleGroup;
+
+    this.buttonToggleGroupMultiple = toggleGroupMultiple;
+
+    if (this.buttonToggleGroup) {
+      buttonToggleDispatcher.listen((id: string, name: string) => {
+        if (id != this.id && name == this.name) {
+          this.checked = false;
+        }
+      });
+
+      this._type = 'radio';
+      this.name = this.buttonToggleGroup.name;
+      this.isSingleSelector = true;
+    } else {
+      // Even if there is no group at all, treat the button toggle as a checkbox so it can be
+      // toggled on or off.
+      this._type = 'checkbox';
+      this.isSingleSelector = false;
+    }
+  }
+
+  /** @internal */
+  ngOnInit() {
+    if (this.id == null) {
+      this.id = `md-button-toggle-${_uniqueIdCounter++}`;
+    }
+
+    if (this.buttonToggleGroup && this._value == this.buttonToggleGroup.value) {
+      this._checked = true;
+    }
+
+  }
+
+  get inputId(): string {
+    return `${this.id}-input`;
+  }
+
+  @HostBinding('class.md-button-toggle-checked')
+  @Input()
+  get checked(): boolean {
+    return this._checked;
+  }
+
+  set checked(value: boolean) {
+    if (value && this.isSingleSelector) {
+      // Notify all button toggles with the same name (in the same group) to un-check.
+      this.buttonToggleDispatcher.notify(this.id, this.name);
+
+      if (!this._checked) {
+        this._emitChangeEvent();
+      }
+    }
+
+    this._checked = value;
+  }
+
+  get type(): ToggleType {
+    return this._type;
+  }
+
+  set type(value: ToggleType) {
+    this._type = value;
+  }
+
+  /** MdButtonToggleGroup reads this to assign its own value. */
+  @Input()
+  get value(): any {
+    return this._value;
+  }
+
+  set value(value: any) {
+    if (this._value != value) {
+      if (this.buttonToggleGroup != null && this.checked) {
+        this.buttonToggleGroup.value = value;
+      }
+      this._value = value;
+    }
+  }
+
+  /** Dispatch change event with current value. */
+  private _emitChangeEvent(): void {
+    let event = new MdButtonToggleChange();
+    event.source = this;
+    event.value = this._value;
+    this.change.emit(event);
+  }
+
+  @HostBinding('class.md-button-toggle-disabled')
+  @Input()
+  get disabled(): boolean {
+    return this._disabled || (this.buttonToggleGroup != null && this.buttonToggleGroup.disabled) ||
+        (this.buttonToggleGroupMultiple != null && this.buttonToggleGroupMultiple.disabled);
+  }
+
+  set disabled(value: boolean) {
+    this._disabled = (value != null && value !== false) ? true : null;
+  }
+
+  get isSingleSelector(): boolean {
+    return this._isSingleSelector;
+  }
+
+  set isSingleSelector(value: boolean) {
+    this._isSingleSelector = value;
+  }
+
+  /** @internal */
+  onClick(event: Event) {
```

Comment:
Changed in a follow-up PR (to onInputChange), had to make other changes to the function anyway + tests so we determined it would be better as a follow-up.